src/client/app/components/groups/EditGroupModalComponent.tsx

```diff
@@ -404,266 +408,292 @@ export default function EditGroupModalComponent(props: EditGroupModalComponentPr
-														const childRemoved = removeChildFromGroup(removedMeterId, DataType.Meter)
-														if (!childRemoved) {
-															// The new child meter removal was rejected so put it back. Should only be one item so no need to sort.
-															newSelectedMeterOptions.push({
-																value: removedMeterId,
-																label: metersState[removedMeterId].identifier
-																// isDisabled not needed since only used for selected and not display.
-															} as SelectOption
-															);
-														}
-													}
-												}}
-											/>
-										</div>
-										:
-										<div>
-											<b><FormattedMessage id='child.meters' /></b>:
-											<ListDisplayComponent items={metersToList()} />
-										</div>
+				<ModalHeader>
+					<FormattedMessage id={loggedInAsAdmin ? 'edit.group' : 'group.details'} />
+					<TooltipHelpContainer page='groups-edit' />
+					<div style={tooltipStyle}>
+						<TooltipMarkerComponent page='groups-edit' helpTextId={tooltipStyle.tooltipEditGroupView} />
+					</div>
+				</ModalHeader>
+				<ModalBody><Container>
+					{loggedInAsAdmin ?
+						<Row xs='1' lg='2'>
+							{/* Name input for admin*/}
+							<Col><FormGroup>
+								<Label for='name'>{translate('group.name')}</Label>
+								<Input
+									id='name'
+									name='name'
+									type='text'
+									autoComplete='on'
+									onChange={e => handleStringChange(e)}
+									required value={groupState.name}
+									invalid={groupState.name === ''}/>
+								<FormFeedback>
+									<FormattedMessage id="error.required" />
+								</FormFeedback>
+							</FormGroup></Col>
+							{/* default graphic unit input for admin */}
+							<Col><FormGroup>
+								<Label for='defaultGraphicUnit'>{translate('group.defaultGraphicUnit')}</Label>
+								<Input
+									id="defaultGraphicUnit"
+									name='defaultGraphicUnit'
+									type='select'
+									value={groupState.defaultGraphicUnit}
+									onChange={e => handleNumberChange(e)}>
+									{/* First list the selectable ones and then the rest as disabled. */}
+									{Array.from(graphicUnitsState.compatibleGraphicUnits).map(unit => {
+										return (<option value={unit.id} key={unit.id}>{unit.identifier}</option>)
+									})}
+									{Array.from(graphicUnitsState.incompatibleGraphicUnits).map(unit => {
+										return (<option value={unit.id} key={unit.id} disabled>{unit.identifier}</option>)
+									})}
+								</Input>
+							</FormGroup></Col>
+						</Row>
+						: <>
+							{/* Name display for non-admin */}
+							<FormGroup>
+								<Label for='name'>{translate('group.name')}</Label>
+								<Input
+									id='name'
+									name='name'
+									type='text'
+									autoComplete='on'
+									value={groupState.name}
+									disabled/>
+							</FormGroup>
+							{/* default graphic unit display for non-admin */}
+							<FormGroup>
```

Comment:
The reason this is because this has to be a select type. Without pulling the entire units state, I'm not sure how to get the identifier of the currently selected graphic unit. There is graphicunitsstate, but that stores a set of UnitData, and I'm not sure how to easily filter that for the identifier I want.

Reply:
The TODO you left is fine.